forem/forem
# Expand Images::Optimizer to support imgproxy

## What type of PR is this? (check all applicable)

## Description
Added Imgproxy support and enable it to the only one area: the mascot that would appear when the user has 0 posts on their dashboard.

This *won't* enable Imgproxy unless both Imgproxy's key and endpoint vars are provided.

## Related Tickets & Documents
https://github.com/orgs/forem/projects/6#card-43004903

## QA Instructions, Screenshots, Recordings
To test out imgproxy, you'll have to do the following

1. You will need to install imgproxy locally. You can do so with `brew install imgproxy`
2. You need to generate a key/salt pair for this to work. In a terminal, rung `echo $(xxd -g 2 -l 64 -p /dev/random | tr -d '\n')` twice and copy those values to your `.env` in the next step
3. In your `.env` set the following
```
export IMGPROXY_ENDPOINT= 'http://localhost:8080'
export IMGPROXY_KEY='1b1c9aae804e070b0864f2547fba7ce8ff31bf7..........'
export IMGPROXY_SALT='8c6d449d4fc2cada5bab538826cae709d2ade9f9286af8b83573346bfb7.........'
```
4. In a separate terminal tab, start imgproxy server by running `IMGPROXY_KEY='your key' IMGPROXY_SALT='your salt' imgproxy`
5. Start rails server normally.
6. Just for this demonstration, please change line 40 of `app/views/dashboards/show.html.erb` to `<% if false %>`. We're pretending that you have 0 articles.
7. Startup rails console on another terminal window, and set your mascot URL with `SiteConfig.mascot_image_url = "https://thepracticaldev.s3.amazonaws.com/i/bh2wmpcltaybu1xsnico.png"`
8. Navigate to `/dashboard`, the dev icon you see should have a link similar to `http://localhost:8080/Utx2iCpXzy8qyckZNCeVL5F7mHtUvUiqEE1gbeys_1g/w:300/aHR0cHM6Ly90aGVw/cmFjdGljYWxkZXYu/czMuYW1hem9uYXdz/LmNvbS9pL2JoMndt/cGNsdGF5YnUxeHNu/aWNvLnBuZw` and not a Cloudinary url.

Please let me know if any of these steps need clarification.
## Added tests?

## Added to documentation?
